src/main/java/io/jenkins/plugins/coverage/model/QualityGate.java

```diff
@@ -0,0 +1,46 @@
+package io.jenkins.plugins.coverage.model;
+
+/**
+ * Stores quality gate limits for specific metric.
+ *
+ * @author Adrian Germeck
+ */
+public class QualityGate {
+    private final CoverageMetric coverageMetric;
+    private final double warningLimit;
+    private final double failedLimit;
+
+    /**
+     * Creates a quality gate.
+     *
+     * @param warningLimit
+     *         coverage percentage of upper border for warning
+     * @param failedLimit
+     *         coverage percentage of upper border for fail
+     * @param coverageMetric
+     *         type of coverage metric
+     */
+    public QualityGate(final double warningLimit, final double failedLimit, final CoverageMetric coverageMetric) {
```

Comment:
This is not exactly what makes sense in our case. Typically one would like to set quality gates for either unstable or failed builds and not both (at least sometimes).